pepper/apps/hashing.h

```diff
@@ -18,12 +21,34 @@ uint32_t hashPedersen(bool *in, uint32_t inputSize, uint32_t chunkSize) {
         decomposeBits(result->values[0], decomposed);
 
         // Fill pedersen input, left with previous result
-        bool pedersenInput[508] = { 0 };
+        bool pedersenInput[PEDERSEN_HASH_SIZE] = { 0 };
         copyBits(decomposed, 0, pedersenInput, 0, 254);
         uint32_t padding = 254 - chunkSize;
         copyBits(in, index, pedersenInput, 254+padding, chunkSize);
 
         ext_gadget(pedersenInput, result, 1);
     }
     return result->values[0];
+}
+
+/**
+ * Hashes the given input of size `inputSize` by dividing it into windows of size 
+ * `chunkSize` and chaining their SHA256 hashes together. It startes with 256 0-bits
+ */
+struct ShaHash { bool digest[256]; };
+void hashSHA(bool *in, uint32_t inputSize, uint32_t chunkSize, int128 result[2]) {
```

Comment:
Hm... working with int128 is fine for now, but later we should switch over to field

Reply:
sure, you made these prs, so that all tests pass. It's good. :+1: